### DISCUSSION TITLE
VE Direkt Anschluss

### DISCUSSION THREAD

**Discussion by Mogdar-M** [hoylabs/OpenDTU-OnBattery#214] [Q&A] [answered]
Hallo,
ich kämpfe damit, den VE Direkt Anschluss herzustellen.
So ist mein Anschluss, um die 5V auf 3,3V zu bekommen nutze ich einen Spannungsteiler
![image](https://user-images.githubusercontent.com/129499239/235424415-a8094d32-1581-4b64-9c6f-e17253df4a2f.png)

Da im Verdrahtungsplan nur EX und TX verbunden sind, der Victron ja keine Daten annimmt habe ich nur TX verbunden.
So ist der Anschlussplan vom Victron 100-20
![image](https://user-images.githubusercontent.com/129499239/235424618-793f044f-9b11-41d0-8ced-33e6df093d5b.png)

Pin 3 im Stecker ist also TX, sendet also die Daten.
Gemäß Plan ist TX an Pin 21 vom ESP, allerdings dachte ich es ist TX vom ESP, daher habe ich den TX vom Victron an Pin 22

Da dies kein Ergenis brachte habe ich also alle voer Varianten ausprobiert:

Pin 2 vom Victron an Pin 21 --> keine Daten
Pin2 vom Victron an Pin 22 --> keine Daten
Pin 3 vom Victron an Pin 21 --> keine Daten
Pin3 vom Victron an Pin 22 --> keine Daten

Der Spannungsteiler ist korrekt berechnet, es sollten unter 10mA und 3,2V am Pin anliegen. Das sollte also gehen.

Was fehlt??

**Comment by helgeerbe** [comment]
Mit einem Spannungsteiler habe ich es auch nie zum laufen gebracht. Mit level Shifter geht es. Zum Testen kannst du esp und Victron auch direkt verbinden (der esp verträgt das). Für den finalen Aufbau würde ich allerdings einen Level Shifter dazwischen setzen.

**Reply by helgeerbe** [reply]
https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/99#discussioncomment-4906584

**Reply by Mogdar-M** [reply]
ok, ich hänge TX und GND mal dran und sehe ob es geht, dann probiere ich ohne Spannungsteiler

**Comment by Mogdar-M** [comment]
Ich kenne da nur die 4 Kanal Dinger, das ist für den einen Anschluss gleich wieder over the top.
Und man bekommt entweder hohe Versandkosten oder gleich viele

**Comment by Mogdar-M** [comment]
btw, ist denn nun Pin3 auf 22 korrekt?
Oder habe ich das falsch verstanden?

**Reply by helgeerbe** [reply, marked as answer]
Victron RX --> ESP32 VICTRON_PIN_TX 
Victron TX --> ESP32 VICTRON_PIN_RX

-DVICTRON_PIN_TX=21
-DVICTRON_PIN_RX=22

Wenn du die default Werte nimmst wäre Pin3 auf 22 korrekt. Ich würde auch das andere Kabel anschliessen. Falls ich doch mal Zeit habe und das Hex-Protokoll implementiere, um auch Daten an den Victron zu senden.

**Comment by Mogdar-M** [comment]
![image](https://user-images.githubusercontent.com/129499239/235464445-2e6ec4ea-4c3b-4ee6-bf1b-2587c6338cc5.png)

Alle drei Kabel direkt angeschlossen, ohne Spannungsteiler und läuft :-)

Danke!

**Reply by dbohne** [reply]
hi, ich kämpfe ebenfalls mit der VE Direct Anbindung. Wärst du so nett und postest  an welche Pins du angeschlossen hast? Ich habe einen ESP32 mit 38 Pins und einen MPPT 100/30. Vielen Dank

**Reply by Mogdar-M** [reply]
Hallo,

ich habe RX und TX auf 21 und 22 angeschlossen und GND dazu.
Seitdem geht es bei mir.

Es kann sein dass beim 100-30 die Pins anders herum sind.
Also miss die äußeren Pins, einer ist 5V, der andere GND.
Die inneren dann RX und TX
